modules/veteran_verification/app/services/bgs/disability_rating_service.rb

```diff
@@ -1,13 +1,59 @@
 # frozen_string_literal: true
 
+require 'savon'
+require 'nokogiri'
+
 module BGS
   class DisabilityRatingService
     def get_rating(current_user)
-      service = BGS::Services.new(
-        external_uid: current_user.icn,
-        external_key: current_user.email
-      )
-      service.rating.find_rating_data(current_user.ssn)
+      if Settings.bgs.mock_bgs == true
+        options = {
+          wsdl: "#{Settings.bgs.url}/RatingServiceBean/RatingService/ratingRecord.wsdl",
+          soap_header: header(current_user),
+          endpoint: "#{Settings.bgs.url}/RatingServiceBean/RatingService"
+        }
+        @client ||= Savon.client(options)
```

Comment:
I think there's too much going on in this section for just mocking a response.  It's also not very reusable for anywhere else we might want to mock BGS.  I think this should be lifted out into it's own class, perhaps `BGS::MockServices`?
So I think it would be neat if this method looked more like this:
```
 def get_rating(current_user)
      service = BGS.services_object(current_user)
      service.rating.find_rating_data(current_user.ssn)
end
```
then there would be a method in the `BGS` module space like this:
```
module BGS
  def self.services_object(current_user)
    if Settings.bgs.mock_bgs
      MockServices.new(current_user)
    else
      Services.new(
        external_uid: current_user.icn,
        external_key: current_user.email
      )
    end
  end
end
```
Then all this logic of instantiating a savon client and building a header and all this stuff can be tucked away in that class and keep this code simple.

Reply:
In the interest of single responsibility classes, I think you're right and I added a commit to define a mock-disability-rating-service separate from disability-rating-service code.  However, to be clear about mock-bgs, it is _only_ for disability rating data at this time - which is specific to veteran verification.  If it were to be expanded to mock other BGS endpoints, that is when we would like to move it to a more global level to prevent misunderstandings.

After discussing with my team (Pivot!) we believe the approach in my latest commit would be the best way to go. We are open to hearing your thoughts though.

Reply:
I'm curious - is there any reason not to sub-namespace the `bgs_disability_rating` service, such as in the `evss:` block of [`settings.yml`](https://github.com/department-of-veterans-affairs/vets-api/blob/master/config/settings.yml#L182)? 

Could be:

```yml
# ...
bgs:
  mock_disability_rating: false
# ...
```

Reply:
Oh I see, it's not data that can be mocked using betamocks, it calls an actual external service... Hmm. I don't think we have an established pattern for this yet, but I do second Philip's comment about tidying up the state